dist/mapbox-gl.css

```diff
@@ -5,6 +5,12 @@
     -webkit-tap-highlight-color: rgba(0,0,0,0);
 }
 
+.mapboxgl-canvas-container {
+    -webkit-transform-style: preserve-3d;
+    -moz-transform-style: preserve-3d;
+    transform-style: preserve-3d;
```

Comment:
Looks like the performance impact is actually relatively minor. I did some tests using `debug/markers3d.html` using 50 markers to check the framerate with different z-Indexing strategies.

No markers (baseline):
![baseline-no-markers](https://cloud.githubusercontent.com/assets/870796/18103069/b0fb3288-6eba-11e6-9217-84a18831a182.png)

Current version in production (ignores zIndexing and scaling completely):
![old-random-ordering](https://cloud.githubusercontent.com/assets/870796/18103100/dbb16998-6eba-11e6-9a40-9aedf2e28829.png)

With translate3d
![ordering-with-translate3d copy](https://cloud.githubusercontent.com/assets/870796/18103121/eed00502-6eba-11e6-8ea0-6f024e0bd224.png)

With zIndex property
![setting-with-zindex](https://cloud.githubusercontent.com/assets/870796/18103132/f7b1ed8e-6eba-11e6-9d8c-f304bdbca814.png)